syntax = "proto3";

service XMLService {

  rpc CsvToXml (CsvRequest) returns (XmlResponse);

  rpc ValidateXml (XmlRequest) returns (ValidationResponse);

  rpc XPathQuery (QueryRequest) returns (QueryResponse);

  rpc XmlInfo (XmlRequest) returns (XmlInfoResponse);

  rpc ExecuteXQuery (XQueryRequest) returns (XQueryResponse);
}

message CsvRequest {
  string csv = 1;
}

message XmlRequest {
  string xml = 1;
}

message ValidationResponse {
  bool valid = 1;
  string message = 2;
}

message QueryRequest {
  string xml = 1;
  string query = 2;
}

message QueryResponse {
  string result = 1;
}

message XmlResponse {
  string xml = 1;
}

message XmlInfoResponse {
  repeated string colunas = 1;
  int32 total = 2;
}

message XQueryRequest {
  string xmlPath = 1;
  string query = 2;
}

message XQueryResponse {
  string result = 1;
}
